src/bar.cpp

```diff
@@ -594,15 +594,10 @@ auto waybar::Bar::setVisible(bool nvis) -> void {
   visible = nvis;
   if (!visible) {
     window.get_style_context()->add_class("hidden");
-    window.set_opacity(0);
   } else {
     window.get_style_context()->remove_class("hidden");
-    window.set_opacity(1);
```

Comment:
You are fixing the symptom and also breaking existing features (show/hide by signal).

`set_opacity` should not fail on a correct visible Gtk window. The only cases of crash I can imagine without looking at the backtrace are when the window is not yet initialized or already destroyed.
Can you try checking `window.get_realized()` or `window.get_mapped()` right after setting the `visible` property?

Reply:
Sorry by introducing breaking changes, that was not my intention. I will reset it to the current master branch implementation.

Reply:
I reset the changes to the setVisible function. Calling `get_realized` or `get_mapped` did not make a difference.

The following error occurs when `set_opacity` is in the function and is used when the modifier key is pressed.
``` 
fish: “./build/waybar” terminated by signal SIGSEGV (Address boundary error)
```